.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -1236,6 +1258,18 @@ private DateTimeResolutionResult ParseYear(string text, DateObject referenceDate
                     if (exactMatch.Success)
                     {
                         year = config.DateExtractor.GetYearFromText(exactMatch.Match);
+                        if (!string.IsNullOrEmpty(exactMatch.Groups["special"].Value))
+                        {
+                            var specialYearTimex = this.config.SpecialYearPrefixesMap[exactMatch.Groups["special"].Value.ToLowerInvariant()];
+
+                            if (!string.IsNullOrWhiteSpace(specialYearTimex))
```

Comment:
Why do we need timex generation code in multiple places? It should at least leverage the same timex generation code.